src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py

```diff
@@ -53,9 +53,39 @@ class MCP2221:
     GP_ALT1 = 0b011
     GP_ALT2 = 0b100
 
-    def __init__(self):
-        self._hid = hid.device()
-        self._hid.open(MCP2221.VID, MCP2221.PID)
+    # attempting to reopen the same hid device results in an error,
+    # so we cache and return already opened devices
+    instances = {}
+
+    def __new__(cls, bus_id=None):
+        """Return a (possibly cached) MCP2221 instance."""
+        print(f'MCP2221.__new__; {bus_id=}; instances={list(cls.instances)}')
+        # check for an existing cached instance
+        if bus_id in cls.instances:
+            # return the corresponding cached instance
+            print(f'Returning cached {bus_id}...')
+            return cls.instances[bus_id]
+        if bus_id is None and cls.instances:
+            # return the instance for the first bus_id, if already cached
+            first_bus_id = cls.available_paths(require_mcps=True)[0]
+            if first_bus_id in cls.instances:
+                print(f'Returning cached {first_bus_id}...')
+                return cls.instances[first_bus_id]
+
+        # create a new instance
+        if bus_id is not None:
+            # use the given bus_id
+            self = super().__new__(cls)
+            self._hid = hid.device()
+            print(f'Opening {bus_id}.')
+            self._hid.open_path(bus_id)
+            self._bus_id = bus_id
+            # cache the new instance
+            cls.instances[bus_id] = self
+        else:
+            # find the first available MCP
+            self = cls.new_instance()
```

Comment:
If no `bus` is specified and the first device is not cached (e.g. because some other process already opened it), then it will try to find and open another one.  This behavior is non-deterministic, since it depends on the order the devices have been connected, and on whether other processes have opened the device.

If we want `MCP2221()` (without the `bus`) to only ever access the first device, the call to `new_instance` can be removed and the code changed to just try to open the first device (and possibly failing and raising an error without falling back on the next available device).

Reply:
I now removed the `new_instance` method and updated the code so that `MCP2221` will try to find and open the first *available* MCP.